docs/odata/doAndDont.markdown

```diff
@@ -0,0 +1,109 @@
+---
+layout: default
+title: Dos and Don'ts
+name: doAndDont
+parent: oData
+nav_order: 4
+---
+
+## Dos and Don'ts
+
+This section contains useful dos and don'ts for developers programming SAP Gateway applications and optimizing performance.
+
+### Dos
+
+#### * Do think in REST terms
+
+If you find yourself inventing function imports such as CreateX, ReadX, GetX, ChangeX, UpdateX, or DeleteX, you should make the entity type X part of your model.
+`CRUD` is covered by `GET`, `PUT`, and `DELETE` to entities and `POST` to entity sets. .
+
+Function imports may be used for operations other than `CRUD`, such as Approve, Reject, Reserve, and Cancel. However, you might find that a reservation entity that you can create, update, and delete is better suited.
+
+#### * Do return completely filled entries in response to GET ~/EntitySet
+
+Think:
+
+```sql
+SELECT * FROM EntitySet WHERE ($filter) ORDER BY ($orderby) UP TO ($top) ROWS
+```
+
+if you see the following:
+
+```sql
+GET ~/EntitySet?$filter=...&$orderby=...&$top=...
+```
+
+OData does not distinguish GetList from GetDetail. OData clients expect to obtain all information from reading an entity set. If the information is not transparent and comprehensive, OData clients might attempt subsequent updates that could lead to errors in your data.
+
+#### * Do use the OData system query options $filter, $top, $skip, $orderby
+
+Clients can, and should, use the query options to tailor responses to your needs. Content creators should support these query options to enable clients to optimize requests.
+
+#### * Do provide a usable set of filterable properties
+
+Make as many properties as possible filterable. At least all primary and foreign key properties should be filterable.
+
+#### * Do make your properties concrete
+
+#### * Do use the right data type for your properties
```

Comment:
might link to example tables "atomic types - odata types" here such as https://help.sap.com/viewer/68bf513362174d54b58cddec28794093/7.51.6/en-US/54a326519eff236ee10000000a445394.html